motion/dp_controller/include/dp_controller/quaternion_dp_controller.h

```diff
@@ -0,0 +1,84 @@
+/*   Written by Kevin Strandenes and Anders Slåkvik, Student
+     Documentation written by Kevin Strandenes and Anders Slåkvik
+     Copyright (c) 2023 Beluga AUV, Vortex NTNU.
+     All rights reserved. */
+
+/**
+ * @file
+ * @brief A nonlinear PID controller for position and orientation
+ *
+ * An implementation of Fjellstad & Fossen 1994: Quaternion
+ * Feedback Regulation of Underwater Vehicles, a nonlinear PD
+ * position and orientation controller. All variables are named
+ * as they appear in the paper.
+ *
+ * Note that the PD controller as described in the paper has been
+ * expanded to a PID controller.
+ */
+
+#ifndef VORTEX_CONTROLLER_QUATERNION_PD_CONTROLLER_H
+#define VORTEX_CONTROLLER_QUATERNION_PD_CONTROLLER_H
+
+#include <Eigen/Dense>
+#include <cmath>
+#include <cstdlib>
+#include <iostream>
+#include <math.h>
+
+#include "eigen_typedefs.h"
+
+class QuaternionPIDController {
+private:
+  float m_W;
+  float m_B;
+  Eigen::Vector3d m_r_B;
+  Eigen::Vector3d m_r_G;
+  Eigen::Vector6d m_p_gain;
+  Eigen::Matrix6d m_i_gain = Eigen::Matrix6d::Zero();
+  Eigen::Vector6d m_d_gain;
+  Eigen::Vector6d integral = Eigen::Vector6d::Zero();
```

Comment:
Why are some of these member variables initialized here, some are initialized in the constructor, and some not at all?